source/extensions/transport_sockets/tls/ocsp/ocsp.cc

```diff
@@ -265,7 +265,14 @@ ResponseData Asn1OcspUtility::parseResponseData(CBS& cbs) {
     throw EnvoyException("OCSP ResponseData is not a well-formed ASN.1 SEQUENCE");
   }
 
-  unwrap(Asn1Utility::skipOptional(elem, CBS_ASN1_CONTEXT_SPECIFIC | CBS_ASN1_CONSTRUCTED | 0));
+  // only support v1, the value of v1 is 0
+  auto version =
+      unwrap(Asn1Utility::getOptional(elem, CBS_ASN1_CONTEXT_SPECIFIC | CBS_ASN1_CONSTRUCTED | 0));
+  if (version.has_value() && *version.value().data != 0) {
+    throw EnvoyException(
+        fmt::format("OCSP ResponseData version {} is not supported", *version.value().data));
+  }
```

Comment:
As I mentioned in the comment on the test, the explicit keyword means getOptional returns a CBS that contains an integer. We need to decode the integer to learn what the value actually is.

```suggestion
  auto version_cbs =
      unwrap(Asn1Utility::getOptional(elem, CBS_ASN1_CONTEXT_SPECIFIC | CBS_ASN1_CONSTRUCTED | 0));
  if (version_cbs.has_value() && version_cbs->len > 0) {
    auto version = unwrap(Asn1Utility::parseInteger(*version_cbs));
    if (version != "00") {
      throw EnvoyException(
          fmt::format("OCSP ResponseData version 0x{} is not supported", version));
    }
  }
```

The parseInteger function returns the value as hex-encoded string (because it can, in theory be an arbitrarily large number). Hence the comparison with "00". I also added `0x` before the version string to make the output clearer. You'll have to update the expected error in the test.